src/contracts/LidoLiquidityManager.sol

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.23;
+
+import {OwnableOperable} from "./OwnableOperable.sol";
+import {IERC20, IWETH, IStETHWithdrawal} from "./Interfaces.sol";
+
+/**
+ * @title Manages stETH liquidity against the Lido Withdrawal Queue.
+ * @author Origin Protocol Inc
+ */
+abstract contract LidoLiquidityManager is OwnableOperable {
+    IERC20 public immutable steth;
+    IWETH public immutable weth;
+    IStETHWithdrawal public immutable withdrawalQueue;
+
+    uint256 public outstandingEther;
+
+    uint256[49] private _gap;
+
+    constructor(address _steth, address _weth, address _lidoWithdrawalQueue) {
+        steth = IERC20(_steth);
+        weth = IWETH(_weth);
+        withdrawalQueue = IStETHWithdrawal(_lidoWithdrawalQueue);
+    }
+
+    /**
+     * @notice Approve the stETH withdrawal contract. Used for redemption requests.
+     */
+    function approveStETH() external onlyOperatorOrOwner {
+        steth.approve(address(withdrawalQueue), type(uint256).max);
+    }
+
+    /**
+     * @notice Request a stETH for ETH withdrawal.
+     * Reference: https://docs.lido.fi/contracts/withdrawal-queue-erc721/
+     * Note: There is a 1k amount limit. Caller should split large withdrawals in chunks of less or equal to 1k each.)
+     */
+    function requestStETHWithdrawalForETH(uint256[] memory amounts)
```

Comment:
`requestStETHWithdrawalForETH` is the name used in the existing OSwap contract for AMM v1. I'm tempted to rename this to something more generic. It will means changing existing off-chain code like the Defender Action and Hardhat task that requests Lido withdrawals.

Reply:
I am fine changing names of privileged-only functions.